pol-core/ecompile/ECompileMain.cpp

```diff
@@ -138,6 +145,14 @@ struct Comparison
 Compiler::SourceFileCache em_parse_tree_cache( summary.profile );
 Compiler::SourceFileCache inc_parse_tree_cache( summary.profile );
 
+using DependencyInfo = std::map<fs::path, std::set<fs::path>>;
+// Map owner (sources) -> dependencies
+DependencyInfo owner_dependencies;
+// Map dependency -> owners (sources)
+DependencyInfo dependency_owners;
+
```

Comment:
You need these globals only because you added code into process_file_wrapper. If you put this code into the calling function these variables can be local.

Reply:
These vars are also used in `add_dependency_info`  as well as `EnterWatchMode`.